### ISSUE TITLE
NTraceGenerator iteration

### ISSUE THREAD

**Issue by dneise** [cta-sst-1m/digicamtoy#7]
`NTraceGenerator` returns itself, might be better to return an event instead of being an event.

**Comment by calispac** [comment]
You mean an event in the sens of `digicampipe` ? if so I don't agree. Might add too much complexity for a random user. You just want the waveform and the input parameters of the MC

**Comment by dneise** [comment]
Ah .. no .. sorry ... I expressed myself badly. At the moment, when iterating over `toy` one receives the instance of `toy`. So `toy` is both the iteration and the result of the iteration. It returns itself.

I wanted to say: I think it is better to **return** the result instead of **being** the result.

This goes in the direction of having a "is-a"-relation or a "has-a"-relation. Like in asking the question:

At the moment, these questions are answered like this:

"Has a `NTraceGenerator` a lot of events, I can iterator over?" - No.
"Is a `NTraceGenerator` a lot of events, I can iterator over?" - Yes.

**Comment by dneise** [comment]
And all this goes in the direction of [Video: Stop-Writing-Classes](https://www.youtube.com/watch?v=o9pEzgHorH0) again...

**Comment by calispac** [comment]
Ok so it should return adc_count of course and also all info related to "True" values.

- cherenkov_time
- cherenkov_photon
- nsb_time
- nsb_photon

**Comment by dneise** [comment]
Yep something like this ...  I mean I'm basically just thinking aloud here in these issues ...
